server/routes/visit.test.ts

```diff
@@ -344,6 +344,21 @@ describe('/visit/:reference', () => {
         })
     })
 
+    it('should display cancel and not the update button if start date has passed by 20 days', () => {
+      const visitDate = new Date(visit.startTimestamp)
+      const testDate = visitDate.setDate(visitDate.getDate() + 20)
```

Comment:
I think this works OK but for here and the test above I wonder if it's better to test closer to the 28 day boundary. E.g. make this test 27 days and the one above 29? (Sort of ignoring 28 itself as when it 'flips over' will depend on the exact visit time...) Could possibly use `date-fns` `addDays() / subDays()` type methods for readability.

Reply:
i found the date-fns add days responded weirdly so kept it as this simple implementation
I did change the tests to 27 and 29 though! Good idea